### ISSUE TITLE
config.local.json is modified by some tests

### ISSUE THREAD

**Issue by benoitchantre** [wprig/wprig#596]
## Issue Overview

When we run `npm run test:gulp:prod:setup`, `config.local.json` is create or modified. As `config.local.json` has precedence over other config files, `npm run bundle` will produce prod files from unexpected settings.

## Steps to Reproduce

1. Create a `config.local.json` with custom values in `config/`
2. Run `npm run test:gulp:prod:setup`
3. See that `config.local.json` has been modified

Content of `config.local.json` after `npm run test:gulp:prod:setup`
```json
{
  "theme": {
    "slug": "gulp-test-prod-theme",
    "name": "Gulp Test Prod Theme",
    "author": "Gulp Bot",
    "PHPNamespace": "Gulp_Bot\\Theme"
  }
}

```

## Expected Behavior

Tests should not modify content or config files.

## Current Behavior

The content of `config.local.json` is modified by `npm run test:gulp:prod:setup`.

**Comment by ataylorme** [comment]
@benoitchantre the tests are primarily meant to be run in the CI service on pull requests.

In that environment, a `config.local.json` must be scaffolded to set the theme name and slug. Without these, the `bundle` command fails.

Scaffolding `config.local.json` was the best way I found to do that as it is not committed to the repo.

can you provide more background on _why_ you are running `npm run test:gulp:prod:setup` locally?

Some possible solutions:

- Check for an existing `config.local.json` and move it to `config.local.old.json` during test set up and revert it during test teardown
- Check for an existing `config.local.json` and do not create one if it exists

I'm leaning towards the first option since the values of an existing `config.local.json` will be unpredictable.

**Comment by benoitchantre** [comment]
Thank you @ataylorme for the details.

I ran a test command while working on #587. I don't think i ran `npm run test:gulp:prod:setup` directly but another command that called it, probably `npm run test:gulp:prod`.

Feel free to close this issue if you think this is an edge case that should not happen during the development of a theme.

**Comment by ataylorme** [comment]
>  I don't think i ran npm run `test:gulp:prod:setup` directly but another command that called it, probably `npm run test:gulp:prod`.

Yes, `npm run test:gulp:prod` runs set up, the test suite, then teardown

> Feel free to close this issue if you think this is an edge case that should not happen during the development of a theme.

The gulp tests should only be run when building WP Rig itself, not when developing a theme based on WP Rig. You can run it locally but should probably do so from a fork of WP Rig itself, on a branch just for the thing you are working on as outlined in [step 2 - 4 of the workflow section in `CONTRIBUTING`](https://github.com/wprig/wprig/blob/master/CONTRIBUTING.md#workflow).

That being said, not having the gulp tests set up clobber an existing `config.local.json` file seems like a good idea

**Comment by ataylorme** [comment]
@benoitchantre can you test/review #605?

**Comment by benoitchantre** [comment]
It works well. Thank you for the fix @ataylorme.

**Comment by benoitchantre** [comment]
Closed by #605.
